### PR TITLE
Migrate to postgres and modify models for consistency

### PR DESCRIPTION
**Pull request by DarkPhoenix42**

### CODE REVIEW

File: api/config.go [hunk 1]
```diff
@@ -162,10 +162,10 @@ func updateChallengeInfoHandler(c *gin.Context) {
 		"Name": name,
 	}
 
-	hints, exist := c.GetPostForm("hints")
-	if exist {
-		configInfo["Hints"] = hints
-	}
+	// hints, exist := c.GetPostForm("hints")
+	// if exist {
```
Comment (sukhman-sukh) [inline api/config.go]: Either uncommment or remove

**Comment by sukhman-sukh** [comment]
Rebase onto bl4ze/dev branch

**Comment by sukhman-sukh** [comment]
PR looks good to me

**Comment by sukhman-sukh** [comment]
Hey @DarkPhoenix42, please resolve the above conflicts to merge this PR.

File: core/config/config.go [hunk 2]
```diff
@@ -305,6 +321,27 @@ func (config *GitRemote) ValidateGitConfig() error {
...
+}
+
+func (config *PsqlConfig) ValidatePsqlConfig() error {
+	if config.User == "" || config.Password == "" || config.Dbname == "" || config.Host == "" || config.Port == "" {
+		log.Error("One of username, password, dbname, hostname, port is missing in the config")
+		return errors.New("psql config not valid, config parameters missing")
+	}
+	if config.SslMode == "" {
+		log.Warn("Ssl Mode not set. Disabling it.")
+		config.SslMode = "disable"
```
Comment (sukhman-sukh) [inline core/config/config.go]: Make it "prefer" by default

File: core/database/database.go [hunk 3]
```diff
@@ -91,37 +118,164 @@ func init() {
...
-		log.Errorf("Error while backing up staging dir: %s", err)
+		log.Printf("Create DB error: %s\n", string(output))
+		return err
+	}
+	log.Debug("Reset successful.")
+	return nil
+}
+
+// Terminate all active connections before dropping
+func TerminateDatabaseConnections() error {
```
Comment (sukhman-sukh) [inline core/database/database.go:222]: Add this as defer when we start beast server for graceful shutdown

**Comment by sukhman-sukh** [comment]
@DarkPhoenix42, Great Work on the migration issue. 
Please fix the above-suggested changes and resolve the conflicts. 
Once that is done, the PR is good to go for merge.

**Comment by sukhman-sukh** [comment]
NVM, I fixed the above issues.

**Comment by sukhman-sukh** [comment]
Also, Did you test it with empty config?
Is it doing validation check before trying to initialize db?
